python/tritonserver/_c/tritonserver_pybind.cc

```diff
@@ -896,107 +871,120 @@ class PyInferenceRequest
+    });
+  }
+  static TRITONSERVER_Error* ResponseAllocatorQueryFn(
+      struct TRITONSERVER_ResponseAllocator* allocator, void* userp,
+      const char* tensor_name, size_t* byte_size,
+      TRITONSERVER_MemoryType* memory_type, int64_t* memory_type_id)
   {
-    request_callback_resource_.reset(
-        new CallbackResource(release, user_object));
-    ThrowIfError(TRITONSERVER_InferenceRequestSetReleaseCallback(
-        triton_object_, PyTritonRequestReleaseCallback,
-        request_callback_resource_.get()));
+    // TODO: Support GPU memory.
+    *memory_type = TRITONSERVER_MEMORY_CPU;
+    *memory_type_id = 0;
+    return nullptr;
   }
-
-  static void PyTritonRequestReleaseCallback(
-      struct TRITONSERVER_InferenceRequest* request, const uint32_t flags,
-      void* userp)
+  static TRITONSERVER_Error* ResponseAllocatorBufferAttributesFn(
+      struct TRITONSERVER_ResponseAllocator* allocator, const char* tensor_name,
+      struct TRITONSERVER_BufferAttributes* buffer_attributes, void* userp,
+      void* buffer_userp)
   {
-    py::gil_scoped_acquire gil;
-    auto cr = reinterpret_cast<CallbackResource*>(userp);
-    cr->release_fn(cr->request, flags, cr->user_object);
-    delete cr;
+    ThrowIfError(TRITONSERVER_BufferAttributesSetMemoryType(
+        buffer_attributes, TRITONSERVER_MEMORY_CPU));
+    ThrowIfError(TRITONSERVER_BufferAttributesSetMemoryTypeId(
+        buffer_attributes, 0 /* memory_type_id */));
+    return nullptr;
+  }
+  static TRITONSERVER_Error* ResponseAllocatorAllocFn(
+      struct TRITONSERVER_ResponseAllocator* allocator, const char* tensor_name,
+      size_t byte_size, TRITONSERVER_MemoryType memory_type,
+      int64_t memory_type_id, void* userp, void** buffer, void** buffer_userp,
+      TRITONSERVER_MemoryType* actual_memory_type,
+      int64_t* actual_memory_type_id)
+  {
+    *buffer = malloc(byte_size * sizeof(uint8_t));
+    *actual_memory_type = TRITONSERVER_MEMORY_CPU;
+    *actual_memory_type_id = 0;
+    return nullptr;
+  }
+  static TRITONSERVER_Error* ResponseAllocatorReleaseFn(
+      struct TRITONSERVER_ResponseAllocator* allocator, void* buffer,
+      void* buffer_userp, size_t byte_size, TRITONSERVER_MemoryType memory_type,
+      int64_t memory_type_id)
+  {
+    free(buffer);
+    return nullptr;
   }
 
-  void SetResponseCallback(
-      const py::object& allocator, const py::object& allocater_user_object,
-      PyInferenceResponse::CompleteFn response,
-      const py::object& response_user_object)
+  void SetResponseCallback()
   {
-    allocator_callback_resource_.reset(
-        new PyResponseAllocator::CallbackResource(
-            allocator, allocater_user_object));
-    response_callback_resource_.reset(new PyInferenceResponse::CallbackResource(
-        response, allocator_callback_resource_.get(), response_user_object));
+    // The caller is responsible for keeping the request wrapper alive until
```

Comment:
this is maintained by the _api ? is that correc t- so user doesn't have to worry about it explicitly -double checking

Reply:
yes, as long as the user is using the response iterators on the _api.